src/account/account.service.ts

```diff
@@ -92,6 +101,59 @@ export class AccountService {
         return this.accountRepository.getByApId(id);
     }
 
+    async ensureByApId(
+        id: URL,
+    ): Promise<Result<Account, RemoteAccountFetchError>> {
+        const account = await this.accountRepository.getByApId(id);
+        if (account) {
+            return ok(account);
+        }
+
+        const context = this.fedifyContextFactory.getFedifyContext();
+
+        const documentLoader = await context.getDocumentLoader({
+            handle: 'index',
+        });
+
+        let actor: Actor;
+
+        try {
+            const potentialActor = await lookupObject(id, { documentLoader });
+
+            if (potentialActor === null) {
+                return error('not-found');
+            }
+
+            if (!isActor(potentialActor)) {
+                return error('invalid-type');
+            }
+
+            actor = potentialActor;
+        } catch (err) {
+            return error('network-failure');
+        }
+
+        let data: ExternalAccountData;
+
+        try {
+            data = await mapActorToExternalAccountData(actor);
+        } catch (err) {
+            return error('invalid-data');
+        }
+
+        await this.createExternalAccount(data);
+
+        const createdAccount = await this.accountRepository.getByApId(id);
+
+        if (!createdAccount) {
+            throw new Error(
```

Comment:
We want to throw here because this is a legitimate what-the-fuck error, it should 500, because something is seriously wrong

Reply:
Similarly we don't wrap the calls to `getByApId` in a try/catch because if these fail - something is in a bad state and we shouldn't continue